app/pkg/dbx/migrate.go

```diff
@@ -140,3 +143,31 @@ func getLastMigration() (int, error) {
 
 	return int(lastVersion.Int64), nil
 }
+
+func getPendingMigrations(versions []int) ([]int, error) {
+	pendingMigrations := make([]int, 0)
+	versionStr := strconv.Itoa(versions[0])
+
+	for _, version := range versions {
+		versionStr = versionStr + "," + strconv.Itoa(version)
+	}
+
+	dbVersionMap := make(map[int]bool)
+	rows, err := conn.Query("SELECT version FROM migrations_history WHERE version IN (" + versionStr + ")")
+	if err != nil {
+		return nil, err
+	}
+	for rows.Next() {
+		var version int
+		_ = rows.Scan(&version)
+		dbVersionMap[version] = true
+	}
```

Comment:
I first had trouble understanding why you used a map "version => boolean" but I understood while writing my suggestion. I don't know about performances, but anyway I think this is not a big deal because the bottleneck of this function is the SQL request.
Here is my original suggestion:

Instead of using a temporary map, you can build the final array by removing versions already applied (found in the database).

You need to start from a copy of all versions:
```go
pendingMigrations := slices.Clone(versions)
```

And then modify the loop as follow:
```suggestion
	for rows.Next() {
		var version int
		_ = rows.Scan(&version)
		i := slices.Index(pendingMigrations, version)
		if i != -1 {
			pendingMigrations = slices.Delete(pendingMigrations, i, i + 1)
		}
	}
```

I think this simplifies a little bit the code

Reply:
I also agree this is more readable, at the beginning, I started to implement the solution using `slices` package and then I imagined that we don't want to import new package on this file, So I managed to find a solution without importing package 😅 

I will refactor this part with `slices` based on your suggestion 👍

Reply:
I'm not a Go specialist, and I don't know the exact impact of importing a package. Because `slices` is part of the standard library, and I guess is already used somewhere else in the project (or in any dependency), I imagine those functions were already included in the final executable. If not, it may be appropriate to do without them, indeed.